python/kserve/kserve/handlers/v1_endpoints.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Union, Dict, List
+
+from fastapi import Request, Response
+from fastapi.responses import ORJSONResponse
+
+from kserve.errors import ModelNotReady
+from kserve.handlers.dataplane import DataPlane
+from kserve.handlers.model_repository_extension import ModelRepositoryExtension
+
+
+class V1Endpoints:
+    """KServe V1 Endpoints
+    """
+
+    def __init__(self, dataplane: DataPlane, model_repository_extension: Optional[ModelRepositoryExtension] = None):
+        self.model_repository_extension = model_repository_extension
+        self.dataplane = dataplane
+
+    async def models(self) -> Dict[str, List[str]]:
+        """Get a list of models in the model registry.
+
+        Returns:
+            Dict[str, List[str]]: List of model names.
+        """
+        return {"models": list(self.dataplane.model_registry.get_models().keys())}
+
+    async def model_ready(self, model_name: str) -> Dict[str, Union[str, bool]]:
+        """Check if a given model is ready.
+
+        Args:
+            model_name (str): Model name.
+
+        Returns:
+            Dict[str, Union[str, bool]]: Name of the model and whether it's ready.
+        """
+        model_ready = self.dataplane.model_ready(model_name)
+
+        if not model_ready:
+            raise ModelNotReady(model_name)
```

Comment:
Raising exception here will return `HTTP 500` according to `model_not_ready_handler`

would be better to just return json response with `ready: False`

Reply:
I also agree with what you said. 
but this behavior is actually compatible with what we had before:  https://github.com/kserve/kserve/blob/master/python/kserve/test/test_server.py#L337
we may need to discuss with Dan what's the desired response.